src/core/server/plugins/plugin_context.ts

```diff
@@ -65,6 +79,23 @@ export function createPluginInitializerContext(
      * Core configuration functionality, enables fetching a subset of the config.
      */
     config: {
+      /**
+       * Global configuration
+       * Note: naming not final here, it will be renamed in a near future (https://github.com/elastic/kibana/issues/46240)
+       * @deprecated
+       */
+      globalConfig__deprecated$: zip(
+        coreContext.configService.atPath<KibanaConfigType>(kibanaConfig.path),
+        coreContext.configService.atPath<ElasticsearchConfigType>(elasticsearchConfig.path),
+        coreContext.configService.atPath<PathConfigType>(pathConfig.path)
```

Comment:
I think we want `combineLatest` instead of `zip` here: if any subconfig changes, we want to reemit from the aggregated `globalConfig__deprecated$`?

Reply:
Good call! Changing it!